src/main/scala/forms/DateWithDaysRule.scala

```diff
@@ -0,0 +1,21 @@
+package forms
+
+import play.api.libs.json.{JsObject, JsValue}
+
+
+class DateWithDaysRule(dateName: String, dateRules: Seq[FieldRule], daysName: String, daysRules: Seq[FieldRule]) extends FieldRule {
+  override def validateOnPreview: Boolean = true
+
+  override def validate(value: JsValue): Seq[String] = {
+    val o = value.validate[JsObject].asOpt.getOrElse(JsObject(Seq()))
+
+    val dateValue =(o \ dateName).toOption.getOrElse(JsObject(Seq()))
+    val daysValue = (o \ daysName).toOption.getOrElse(JsObject(Seq()))
+
+    //dateRules.flatMap()
+
+    ???
```

Comment:
It throws a NotImplementedError. It's a quick way of stubbing out a function that isn't complete so that it throws an error when you run the unit test.

In fact, the FieldRules are no more so I'll delete that class.